eclass/qt5-build.eclass

```diff
@@ -83,15 +94,25 @@ _QT5_P=${QT5_MODULE}-everywhere-src-${PV}
 inherit estack flag-o-matic toolchain-funcs virtualx
 
 if [[ ${PN} != qtwebengine ]]; then
-	if [[ ${QT5_BUILD_TYPE} == live ]] || [[ -n ${KDE_ORG_COMMIT} ]]; then
-		# KDE Qt5PatchCollection
-		inherit kde.org
-	else
-		# official stable release
-		HOMEPAGE="https://www.qt.io/"
-		SRC_URI="https://download.qt.io/official_releases/qt/${PV%.*}/${PV}/submodules/${_QT5_P}.tar.xz"
-		S=${WORKDIR}/${_QT5_P}
-	fi
+	case ${PV} in
+		5.15.2* | *9999 )
+			# kde/5.15 branch or Qt5PatchCollection based on Qt 5.15.2
+			# (using KDE_ORG_COMMIT)
+			inherit kde.org
+			;;
+		5.15.[3-9]* )
+			# official stable release
+			_QT5_P=${QT5_MODULE}-everywhere-opensource-src-${PV}
+			HOMEPAGE="https://www.qt.io/"
+			SRC_URI="https://download.qt.io/official_releases/qt/${PV%.*}/${PV}/submodules/${_QT5_P}.tar.xz"
+			# KDE Qt5PatchCollection on top of tag v${PV}-lts-lgpl
+			if [[ -n ${QT5_KDEPATCHSET_NO} ]]; then
+				HOMEPAGE+=" https://invent.kde.org/qt/qt/${QT5_MODULE} https://community.kde.org/Qt5PatchCollection"
+				SRC_URI+=" https://dev.gentoo.org/~asturm/distfiles/${QT5_MODULE}-${PV}-gentoo-kde-${QT5_KDEPATCHSET_NO}.tar.xz"
+			fi
+			S="${WORKDIR}"/${_QT5_P/opensource-/}
```

Comment:
(nit) quoting technically unnecessary but I don't particularly care :)
and the second `/` in the pattern replacement can be omitted as well